src/EditorFeatures/Core/Intents/IntentEditsSource.cs

```diff
@@ -0,0 +1,86 @@
+using Microsoft.CodeAnalysis.Shared.Extensions;
+using Microsoft.CodeAnalysis.Text;
+using Roslyn.Utilities;
+
+namespace Microsoft.CodeAnalysis.Editor.Intents
+{
+    [Export(typeof(IIntentsEditsSource)), Shared]
+    internal class IntentEditsSource : IIntentsEditsSource
+    {
+        private readonly ImmutableDictionary<(string LanguageName, string IntentName), Lazy<IIntentProvider, IIntentProviderMetadata>> _lazyIntentProviders;
+
+        [ImportingConstructor]
+        [Obsolete(MefConstruction.ImportingConstructorMessage, error: true)]
+        public IntentEditsSource([ImportMany] IEnumerable<Lazy<IIntentProvider, IIntentProviderMetadata>> lazyIntentProviders)
+        {
+            _lazyIntentProviders = CreateProviderMap(lazyIntentProviders);
+        }
+
+        private static ImmutableDictionary<(string LanguageName, string IntentName), Lazy<IIntentProvider, IIntentProviderMetadata>> CreateProviderMap(IEnumerable<Lazy<IIntentProvider, IIntentProviderMetadata>> lazyIntentProviders)
+        {
+            return lazyIntentProviders.ToImmutableDictionary(provider => (provider.Metadata.LanguageName, provider.Metadata.IntentName), provider => provider);
+        }
+
+        public async Task<IntentResult?> ComputeEditsAsync(IntentRequestContext intentRequestContext, CancellationToken cancellationToken)
+        {
+            var originalDocument = intentRequestContext.SnapshotBeforeIntent.GetOpenDocumentInCurrentContextWithChanges();
+            var currentDocument = intentRequestContext.CurrentSnapshot.GetOpenDocumentInCurrentContextWithChanges();
+
+            Contract.ThrowIfNull(originalDocument);
+            Contract.ThrowIfNull(currentDocument);
+            Contract.ThrowIfFalse(originalDocument.Id == currentDocument.Id);
+
+            var languageName = originalDocument.Project.Language;
+            if (!_lazyIntentProviders.TryGetValue((LanguageName: languageName, IntentName: intentRequestContext.IntentName), out var provider))
+            {
+                Logger.Log(FunctionId.Intellicode_UnknownIntent, KeyValueLogMessage.Create(LogType.UserAction, m =>
+                {
+                    m["intent"] = intentRequestContext.IntentName;
+                    m["language"] = languageName;
+                }));
+
+                return null;
+            }
+
+            var selectionTextSpan = intentRequestContext.Selection.ToTextSpan();
+            var newSolution = await provider.Value.ComputeActionForIntentAsync(originalDocument, selectionTextSpan, currentDocument, intentRequestContext.IntentData, cancellationToken).ConfigureAwait(false);
+            if (newSolution == null)
+            {
+                return null;
+            }
+
+            // Merge linked file changes so all linked files have the same text changes.
+            newSolution = await newSolution.WithMergedLinkedFileChangesAsync(originalDocument.Project.Solution, cancellationToken: cancellationToken).ConfigureAwait(false);
```

Comment:
annoyed that you have to do this.  what's the consequence if we dont'?  i feel like as we're not applying to the solution, we likely don't need to change this ourselves.

Reply:
I don't think it applies to generate ctor specifically (I couldn't think of a scenario anyway), but for other code actions it definitely could

e..g renaming
```
class Class2
{
    int someField = 1;

    void M()
    {
        #if NETCOREAPP3_1
        someField++;
        #endif

        #if NETSTANDARD2_0
        someField--;
        #endif
    }
}
```
needs to account for changes in both documents